assets/js/layouts/playground.js

```diff
@@ -26,12 +26,17 @@ export class Playground {
 		return doc;
 	}
 
-	async switchTo(link) {
+	async switchTo(link, target) {
 		// fade out the old image
-		this.$el
-			.querySelector(".playground-header-figure-wrapper")
+		this.currentWrapper()
 			.classList.add("loading");
 
+		// preload the new image
+		const imgSource = this.currentImg()
+			.currentSrc
+			.replace(this.currentWrapper().dataset.image, target.dataset.image);
+		new Image().src = imgSource;
```

Comment:
`Suggestion:`

```suggestion
		new Image().src = this.image.currentSrc.replace(
			this.wrapper.dataset.image, 
			target.dataset.image
		);
```

Depends on suggested change to getters, see below.

`Suggestion:` Could be moved to a separate method `preload(image)`.